packages/flutter/test/gestures/scale_test.dart

```diff
@@ -704,6 +704,461 @@ void main() {
+      updatedScale = details.scale;
+      updatedHorizontalScale = details.horizontalScale;
+      updatedVerticalScale = details.verticalScale;
+      updatedFocalPoint = details.focalPoint;
+      updatedRotation = details.rotation;
+      updatedDelta = details.focalPointDelta;
+    };
+
+    bool didEndScale = false;
+    scale.onEnd = (ScaleEndDetails details) {
+      didEndScale = true;
+    };
+
+    final TestPointer pointer1 = TestPointer(2);
+
+    final PointerPanZoomStartEvent start = pointer1.panZoomStart(Offset.zero);
+    scale.addPointerPanZoom(start);
+    drag.addPointerPanZoom(start);
+
+    tester.closeArena(2);
+    expect(didStartScale, isFalse);
+    expect(updatedScale, isNull);
+    expect(updatedFocalPoint, isNull);
+    expect(updatedDelta, isNull);
+    expect(didEndScale, isFalse);
+
+    tester.route(start);
+    expect(didStartScale, isFalse);
+    expect(updatedScale, isNull);
+    expect(updatedFocalPoint, isNull);
+    expect(updatedDelta, isNull);
+    expect(didEndScale, isFalse);
+
+    // Zoom enough to win the gesture
+    tester.route(pointer1.panZoomUpdate(Offset.zero, scale: 1.1, rotation: 1));
+    expect(didStartScale, isTrue);
+    didStartScale = false;
+    expect(updatedFocalPoint, Offset.zero);
+    updatedFocalPoint = null;
+    expect(updatedScale, 1.0);
+    updatedScale = null;
+    expect(updatedDelta, Offset.zero);
+    updatedDelta = null;
+    expect(didEndScale, isFalse);
+
+    // Zoom in - should be relative to 1.1
+    tester.route(pointer1.panZoomUpdate(Offset.zero, scale: 1.21, rotation: 1.5));
+    expect(updatedFocalPoint, Offset.zero);
+    updatedFocalPoint = null;
+    expect(updatedScale, closeTo(1.1, 0.0001));
+    expect(updatedHorizontalScale, closeTo(1.1, 0.0001));
+    expect(updatedVerticalScale, closeTo(1.1, 0.0001));
+    expect(updatedRotation, 0.5);
+    expect(updatedDelta, Offset.zero);
+    updatedScale = null;
+    updatedHorizontalScale = null;
+    updatedVerticalScale = null;
+    updatedRotation = null;
+    updatedDelta = null;
+    expect(didEndScale, isFalse);
+
+    // Zoom out - should be relative to 1.1
+    tester.route(pointer1.panZoomUpdate(Offset.zero, scale: 0.99, rotation: 1.0));
+    expect(updatedFocalPoint, Offset.zero);
+    updatedFocalPoint = null;
+    expect(updatedScale, closeTo(0.9, 0.0001));
+    expect(updatedHorizontalScale, closeTo(0.9, 0.0001));
+    expect(updatedVerticalScale, closeTo(0.9, 0.0001));
+    expect(updatedRotation, 0.0);
+    expect(updatedDelta, Offset.zero);
+    updatedScale = null;
+    updatedHorizontalScale = null;
+    updatedVerticalScale = null;
+    updatedDelta = null;
+    expect(didEndScale, isFalse);
+
+    // We are done
+    tester.route(pointer1.panZoomEnd());
+    expect(didStartScale, isFalse);
+    expect(updatedFocalPoint, isNull);
+    expect(updatedScale, isNull);
+    expect(updatedDelta, isNull);
+    expect(didEndScale, isTrue);
+    didEndScale = false;
+
+    scale.dispose();
+  });
+
```

Comment:
Is rotation tested anywhere in the tests that you added?

Reply:
Nit: Also, do you think it would be useful to add a `testWidgets` test for this in order to test everything end-to-end in a more real world example?

Reply:
In 'Pointer pan/zooms should work alongside touches', it's tested that the rotation from trackpad gestures is passed along and combined if necessary with any touch pointers. 
Not against writing one more test if you think it's needed but I'm not sure what would be in the `testWidgets`.

Reply:
I see it now, I think that's fine as-is 👍